controllers/nodemaintenance_controller_test.go

```diff
@@ -2,338 +2,367 @@ package controllers
+				time.Sleep(1 * time.Second)
+
+				Expect(k8sClient.Get(context.Background(), client.ObjectKey{Name: nm.Spec.NodeName}, node)).NotTo(HaveOccurred())
+				_, exist := node.Labels[commonLabels.ExcludeFromRemediation]
+				Expect(exist).To(BeFalse())
+			})
 		})
-
-		It("should fail on non existing node", func() {
-			nmFail := getTestNM()
-			nmFail.Spec.NodeName = "non-existing"
-			err := k8sClient.Delete(context.TODO(), nm)
-			Expect(err).NotTo(HaveOccurred())
-			err = k8sClient.Create(context.TODO(), nmFail)
-			Expect(err).NotTo(HaveOccurred())
-			reconcileMaintenance(nm)
-			checkFailedReconcile()
+		When("nm CR is invalid for a missing node", func() {
+			BeforeEach(func() {
+				nm = getTestNM("non-existing-node-cr", invalidNodeName)
+				Expect(k8sClient.Create(context.Background(), nm)).To(Succeed())
+				DeferCleanup(k8sClient.Delete, context.Background(), nm)
+			})
+			It("should fail on non existing node", func() {
+				By("check nm CR status and whether LastError was updated")
+				maintenance := getNMAfter1Sec(nm)
+				Expect(maintenance.Status.Phase).To(Equal(nodemaintenanceapi.MaintenanceRunning))
+				Expect(len(maintenance.Status.PendingPods)).To(Equal(0))
+				Expect(maintenance.Status.EvictionPods).To(Equal(0))
+				Expect(maintenance.Status.TotalPods).To(Equal(0))
+				Expect(maintenance.Status.DrainProgress).To(Equal(0))
+				Expect(maintenance.Status.LastError).To(Equal(fmt.Sprintf("nodes \"%s\" not found", invalidNodeName)))
+				Expect(maintenance.Status.LastUpdate.IsZero()).To(BeFalse())
+				Expect(maintenance.Status.ErrorOnLeaseCount).To(Equal(0))
+			})
 		})
-
-		It("add/remove Exclude remediation label", func() {
-			reconcileMaintenance(nm)
-			checkSuccesfulReconcile()
-			node := &corev1.Node{}
-			err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: nm.Spec.NodeName}, node)
-			Expect(err).NotTo(HaveOccurred())
-			//Label added on CR creation
-			Expect(node.Labels[labels.ExcludeFromRemediation]).To(Equal("true"))
-
-			Expect(k8sClient.Delete(context.Background(), nm)).To(Succeed())
-			reconcileMaintenance(nm)
-			//Re-fetch node after reconcile
-			Expect(k8sClient.Get(context.TODO(), client.ObjectKey{Name: nm.Spec.NodeName}, node)).NotTo(HaveOccurred())
-			_, exist := node.Labels[labels.ExcludeFromRemediation]
-			Expect(exist).To(BeFalse())
-		})
-
 	})
 })
 
-func getTestObjects() (*nodemaintenanceapi.NodeMaintenance, []client.Object) {
-	nm := getTestNM()
+func getNodeMaintenance(nm *nodemaintenanceapi.NodeMaintenance, timeout, pollTime string) *nodemaintenanceapi.NodeMaintenance {
+	maintenance := &nodemaintenanceapi.NodeMaintenance{}
+	Consistently(func() error {
+		return k8sClient.Get(context.Background(), client.ObjectKeyFromObject(nm), maintenance)
+	}, timeout, pollTime).Should(BeNil())
+	return maintenance
+}
+func getNMAfter1Sec(nm *nodemaintenanceapi.NodeMaintenance) *nodemaintenanceapi.NodeMaintenance {
```

Comment:
hm.... what are we doing here? Why not just sleeping for a second? What's the value of getting the CR 5 times very 200ms?